src/AutoRest.CSharp/LowLevel/Generation/LowLevelConvenienceMethodWriter.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using AutoRest.CSharp.Common.Generation.Writers;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Shared;
+using Azure;
+
+namespace AutoRest.CSharp.Generation.Writers
+{
+    internal class LowLevelConvenienceMethodWriter : ClientWriter
+    {
+        internal static void WriteClientConvenienceMethod(CodeWriter writer, LowLevelConvenienceMethod convenienceMethod, string diagnosticsPropertyName, bool async)
```

Comment:
I suggest to make those method non-static. You can put some common parameters like `CodeWriter writer` as the properties of this class, and then initialize them in the constructor. See `MgmtClientBaseWriter`.
That can greatly improve the codes readability.